### PR TITLE
ibi mgmt: add deploy test suite

### PR DESCRIPTION
**Pull request by trewest**
Add first IBI connected deployment test case

### CODE REVIEW

File: tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+					APIClient, host, MGMTConfig.Cluster.Info.ClusterName, v1.SecretTypeOpaque).WithData(map[string][]byte{
+					"username": []byte(info.BMC.User),
+					"password": []byte(info.BMC.Password),
+				}).Create()
+				Expect(err).NotTo(HaveOccurred(), "error creating bmh secret")
+
+				By("Create baremetalhost for " + host)
+				hostBMH := bmh.NewBuilder(
+					APIClient, host, MGMTConfig.Cluster.Info.ClusterName, info.BMC.URLv4, host, info.BMC.MACAddress, "UEFI")
+				hostBMH.Definition.Spec.AutomatedCleaningMode = "disabled"
```
Comment (achuzhoy) [inline tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go:78]: do we on purpose want to keep this one hardcoded?

File: tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+				Expect(err).NotTo(HaveOccurred(), "error creating baremetalhost")
+			}
+
+			By("Create cluster deployment for IBI installation")
+			_, err = hive.NewABMClusterDeploymentBuilder(
+				APIClient, MGMTConfig.Cluster.Info.ClusterName,
+				MGMTConfig.Cluster.Info.ClusterName, MGMTConfig.Cluster.Info.ClusterName,
+				MGMTConfig.Cluster.Info.BaseDomain, MGMTConfig.Cluster.Info.ClusterName, metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"dummy": "label",
```
Comment (achuzhoy) [inline tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go:90]: do we use this label?

File: tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+					},
+				}).WithPullSecret(MGMTConfig.Cluster.Info.ClusterName).Create()
+			Expect(err).NotTo(HaveOccurred(), "error creating cluster deployment")
+
+			var snoNodeName string
+
+			for hostname := range MGMTConfig.Cluster.Info.Hosts {
+				snoNodeName = hostname
+
+				break
```
Comment (achuzhoy) [inline tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go:100]: Hmm, do we have to use loop (and break) here?

File: tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+				}
+
+				condition, err := imageClusterInstall.GetCompletedCondition()
+				if err != nil {
+					return false, err
+				}
+
+				return condition.Status == "True" && condition.Reason == ibiv1alpha1.InstallSucceededReason, nil
+
+			}).WithTimeout(time.Minute*20).WithPolling(time.Second*5).Should(
```
Comment (achuzhoy) [inline tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go:133]: 20 mins? this includes the installation of the spoke, right?

File: tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go (hunk 1 above)
Comment (trewest) [reply tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go:78]: Yeah need to add it to eco-goinfra but this works for now

File: tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go (hunk 3 above)
Comment (trewest) [reply tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go:100]: It's just an easy way to get the SNO node's name until we support multinode

File: tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go (hunk 4 above)
Comment (trewest) [reply tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go:133]: This is for the full installation of the spoke, yes

File: tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+			splitOCPVersion := strings.Split(MGMTConfig.SeedClusterInfo.SeedClusterOCPVersion, ".")
+
+			if len(splitOCPVersion) < 2 {
+				Skip("Could not determine short OCP version from seed cluster ocp version")
+			}
+
+			ibiImageSetName = strings.Join(splitOCPVersion[:2], ".")
+		})
+
+		It("is successful in connected environment", func() {
```
Comment (achuzhoy) [inline tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go:49]: what "is successful" mean here?

File: tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+			tsparams.ReporterNamespacesToDump[MGMTConfig.Cluster.Info.ClusterName] = "spoke namespace"
+
+			By("Create namespace for IBI installation")
+			_, err := namespace.NewBuilder(APIClient, MGMTConfig.Cluster.Info.ClusterName).Create()
+			Expect(err).NotTo(HaveOccurred(), "error creating namespace")
+
+			By("Get pull secret from cluster")
```
Comment (achuzhoy) [inline tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go:57]:
```suggestion
			By("Get pull secret from hub cluster")
```

File: tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go (hunk 5 above)
Comment (trewest) [reply tests/lca/imagebasedinstall/mgmt/deploy/tests/e2e-deploy-test.go:49]: "Performing image based installation is successful" - I'm going to update it to something more specific though

**Review by achuzhoy — CHANGES_REQUESTED** [review]

**Review by achuzhoy — APPROVED** [review]

**Pull request merged**